src/window/macos.rs

```diff
@@ -0,0 +1,138 @@
+use icrate::{
+    AppKit::{
+        NSColor, NSEvent, NSView, NSViewMinYMargin, NSViewWidthSizable, NSWindow,
+        NSWindowStyleMaskTitled,
+    },
+    Foundation::{MainThreadMarker, NSPoint, NSRect, NSSize},
+};
+use objc2::{declare_class, msg_send_id, mutability::InteriorMutable, rc::Id, ClassType};
+
+use csscolorparser::Color;
+use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
+use winit::window::Window;
+
+use crate::{cmd_line::CmdLineSettings, error_msg, frame::Frame, settings::SETTINGS};
+
+use super::WindowSettings;
+
+declare_class!(
+    // A view to simulate the double-click-to-zoom effect for `--frame transparency`.
+    struct TitlebarClickHandler;
+
+    unsafe impl ClassType for TitlebarClickHandler {
+        type Super = NSView;
+        type Mutability = InteriorMutable;
+        const NAME: &'static str = "TitlebarClickHandler";
+    }
+
+    unsafe impl TitlebarClickHandler {
+        #[method(mouseDown:)]
+        unsafe fn mouse_down(&self, event: &NSEvent) {
+            if event.clickCount() == 2 {
+                self.window().unwrap().zoom(Some(self));
+            }
+        }
+    }
+);
+
+impl TitlebarClickHandler {
+    pub fn new(_mtm: MainThreadMarker) -> Id<TitlebarClickHandler> {
+        unsafe { msg_send_id![Self::alloc(), init] }
+    }
+}
+
+lazy_static! {
+    // This height is in dpi-independent length, convert it to pixel length by multiply with scale factor.
```

Comment:
Slightly confusing wording imo.

```suggestion
    // This height is in dpi-independent length, convert it to pixel length by multiplying it with scale factor.
```